mlx/fast_primitives.h

```diff
@@ -1,5 +1,6 @@
 // Copyright © 2024 Apple Inc.
 
+#include <map>
```

Comment:
nit: also not needed?

Reply:
Instead of removing I ended up switching the `unordered_map`s back to `map`s since it's useful to have ordering when constructing the signature and it seems unlikely to make a big difference performance wise.

Reply:
Sorry I should have thought of this. But it would be better to use an `unordered_map` if it's not too tricky, because we have bindings for it in [MLX C already](https://github.com/ml-explore/mlx-c/blob/main/mlx/c/private/map.h). Otherwise we'll have to bind an ordered and an unordered map there.